src/cargo/core/compiler/job_queue.rs

```diff
@@ -105,12 +105,12 @@ impl<'a> JobState<'a> {
             .send(Message::BuildPlanMsg(module_name, cmd, filenames));
     }
 
-    pub fn stdout(&self, stdout: &str) {
-        drop(self.tx.send(Message::Stdout(stdout.to_string())));
+    pub fn stdout(&self, stdout: String) {
```

Comment:
Heh I felt quite a few times that this change needed to be made, just never worked up enough motivation to get it done :)